### ISSUE TITLE
Tabs Accessibility

### ISSUE THREAD

**Issue by coreymcg** [NAEPDEV/benchmark#9]
**Context**

- The current implementation would be considered a "rough draft".
- Some [unit tests have been added](https://github.com/NAEPDEV/benchmark/blob/master/packages/benchmark/src/components/Tabs/Tabs.test.js) to prevent regressions. More tests still need to be added.
- Both Axe and Accessibility insights have both been run.
- Themes will be added at a later date.

**Considerations**

- Applies the  'tablist', 'tab' and 'tabpanel' roles to their respective elements
- Applies an `aria-label` to the tabslist element if a "label" prop has been passed to the Tabs component
- Sets `aria-selected="true"` attribute to the selected tab and `aria-selected="false"` to the unselected tabs
- Sets a `tabindex` of "0" on the selected tab and "-1" unselected tabs
- `aria-labelledby` and `aria-controls` have been applied to link the tabs with the panels
- Individual tabs can be disabled
- Sets a tabindex of "0" on the tab panel when active. This allows us to focus the panel when navigated to by the down arrow key.
- The focus styling on the tab panel (when using down arrow) is currently the browser default. This is likely to change at some point in the future when we do a system wide review of focus styling.
- When using the arrow keys to navigate the tabs it does not cycle back to the start when navigating right on the last tab

**Keyboard Use**

- The `LeftArrow` and `RightArrow` keys will navigate through the tabs
    - They will skip disabled tabs
- The `DownArrow` key will focus the active Tab Panel
- The `Home` key will go to the first tab
- The `End` key will go to the last tab

**Questions**

- For performance, only the active tab panel is rendered. Do all panels need to be rendered to the DOM?
- When the first tab is disabled what should the home key do?
- When the last tab is disabled what should the end key do?

### References

- [https://www.w3.org/TR/wai-aria-practices/examples/tabs/tabs-1/tabs.html](https://www.w3.org/TR/wai-aria-practices/examples/tabs/tabs-1/tabs.html)
- [https://inclusive-components.design/tabbed-interfaces/](https://inclusive-components.design/tabbed-interfaces/)

**Comment by jimmy-james** [comment]
> **Questions**
> 
> * For performance, only the active tab panel is rendered. Do all panels need to be rendered to the DOM?
I think on initial load, only the first active tab panel should be rendered. The rest should lazy load on click or navigation. Once rendered, however, they should remain on the DOM for quick access. Thoughts?

> * When the first tab is disabled what should the home key do?
> * When the last tab is disabled what should the end key do?

My first thought (which may be wrong in terms of convention) was that Home goes to the first active tab after the disabled first tab. Same with End: go to the second to last active tab before the disabled last tab. If only a single active tab left, Home and End both only go to that tab. Do this until all tabs are disabled.
Another solution would be to use the number pad to navigate to existing tabs (ctrl + number): 1-9. If tab length > 9, then 9 represents always represents the last tab.

**Comment by coreymcg** [comment]
**Tab Content Hiding**
If there are landmarks in the tab content then it may be useful to have the content in the DOM. For example could be a landmark on a final "Review" tab. However, I haven't seen a specific use case for this so we should leave it out until further notice.

**Home & End Keys**
This sounds totally reasonable. Lets go ahead and implement this. 

**Number Key Nav**
I'm on the fence with this one. It's unlikely but there's a chance that this could conflict with parts of the UI. We'll leave this out for now and revisit it when we have more of a complete system.
